connector_salesforce/salesforce_contact/importer.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Nicolas Bessi
+#    Copyright 2014 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import logging
+from openerp.addons.connector.exception import MappingError
+from openerp.addons.connector.unit.mapper import mapping, only_create
+from ..backend import salesforce_backend
+from ..unit.importer_synchronizer import (SalesforceDelayedBatchSynchronizer,
+                                          SalesforceDirectBatchSynchronizer,
+                                          SalesforceImportSynchronizer,
+                                          import_record)
+from ..unit.rest_api_adapter import SalesforceRestAdapter
+from ..unit.mapper import AddressMapper
+_logger = logging.getLogger('salesforce_connector_contact_import')
+
+
+@salesforce_backend
+class SalesforceContactImporter(SalesforceImportSynchronizer):
+    _model_name = 'connector.salesforce.contact'
+
+    def _before_import(self):
+        """Hook called before importing a contact
+        Ensure that related account is present
+        if not it is imported
+        """
+        assert self.salesforce_record
+        with self.session.change_context({'active_test': False}):
+            account_id = self.session.search(
+                'connector.salesforce.account',
+                [('salesforce_id', '=', self.salesforce_record['AccountId'])]
+            )
+        if not account_id:
+            import_record(
+                self.session,
+                'connector.salesforce.account',
+                self.backend_record.id,
+                self.salesforce_record['AccountId']
+            )
```

Comment:
Agreed it is part of the TODO in the README. I will do it in a other PR when I have some time to improve the details